dtests/src/test/scala/io/snappydata/hydra/cdcConnector/CDCValidationApp.scala

```diff
@@ -0,0 +1,221 @@
+            pw.println("=====================================================================================================================")
+            pw.println("For table " + tableName + " Present sqlServer cnt = " + sqlServerTableCnt + " Previous sqlServer cnt " + sqlServerTableCnt1Arr(j))
+            pw.println("For table " + tableName + " Present snappy cnt = " + snappyTableCnt + " Previous snappy cnt " + snappyTableCnt1Arr(j))
+            val sqlServerCntDiff = abs(sqlServerTableCnt - sqlServerTableCnt1Arr(j).toLong)
+            val snappyCntDiff = abs(snappyTableCnt - snappyTableCnt1Arr(j).toLong)
+
+            if (sqlServerCntDiff == snappyCntDiff) {
+              pw.println(s"SUCCESS :$tableName in snappy cluster  has equal difference of $snappyCntDiff before and after ingestion ")
+              pw.println(s"SUCCESS :$tableName in sqlServer cluster  has equal difference of $sqlServerCntDiff before and after ingestion")
+            }
+            else {
+              pw.println("FAILURE:" + tableName + " table has incorrect number of records: sqlServer table has "
+                  + sqlServerTableCnt + " and snappytable has " + snappyTableCnt)
+            }
+            pw.println("=====================================================================================================================")
+            pw.println()
+          }
+        }
+      }
+
+       // For printing the map values.
+      def printMapValues(arr : Array[String]): Unit = {
+        pw1.println(s"$arr values are : ")
+        for(i <-0 to arr.length - 1){
+          val value = arr(i)
+          pw1.println(s"$value")
+        }
+      }
+
+      //validation in case of updates:
+      def checkDataConsistency(): Unit = {
+        // pw.println("Inside checkDataConsistency function")
+        for (i <- 0 to tableNameArr.length - 1) {
+          val tableName = tableNameArr(i)
+          val sqlServerQ = s"SELECT * FROM [testdatabase].[dbo].[$tableName]"
+          pw.println("Query is " + sqlServerQ)
+          val sqlServerResultSet = connection.createStatement().executeQuery(sqlServerQ)
+          val snappyDF = snc.sql(s"SELECT * FROM $tableName")
+          //.collect()
+          var cnt = 0
+          val colName = idValArr(i)
+
+          // List of snappy rows for a table.
+          val snappyList = snappyDF.select(s"$colName").collectAsList()
+          val snappArr = snappyList.toArray
+          pw.println("List size = " + snappArr.length + " value is " + snappArr(i))
+
+          val sqlServerIdValArr: Array[String] = new Array[String](snappyDF.count().toInt)
+
+          while (sqlServerResultSet.next()) {
+            sqlServerIdValArr(cnt) = sqlServerResultSet.getString(colName)
+            cnt = +1
+          }
+          pw.println("sqlServerIdValArr size = = " + sqlServerIdValArr.length)
+
+          val diff = sqlServerIdValArr.toSet.diff(snappArr.toSet)
+
+          pw.println("The difference is " + diff)
```

Comment:
Fail the test in case of validation failure (diff id observed).

Reply:
Added exception.